.github/workflows/build.yaml

```diff
@@ -0,0 +1,199 @@
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    types: [opened, synchronize, reopened]
+  release:
+    types:
+    - created
+
+env:
+  MASTER_BRANCH: master
+  BUILD_HARNESS_VERSION: 0.5.5
+  HUGO_CONFIG: deploy.toml
+  HTMLTEST_CONFIG: .htmltest.codefresh.yml
+  TMPDIR: ${{ github.workspace }}
+  YARN_BUILD_DISABLED: false
+  COMPONENTS_BUILD: true
+
+jobs:
+  hugo-build:
+    name: "Build Hugo Site"
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Checkout source code at current commit"
+      uses: actions/checkout@v2
+    - name: "Set up Docker Buildx"
+      uses: docker/setup-buildx-action@v1
+    - name: "Build docker image"
+      uses: docker/build-push-action@v2
+      with:
+        tags: cloudposse/docs
+        outputs: type=docker,dest=/tmp/image.tar
+    - name: "Import Image"
+      run: |
+        ls -lhrta /tmp
+        ls -lhrta /tmp/image.tar
+        docker load --input /tmp/image.tar
+    - name: "Build Hugo"
+      run: |
+        set +e -x
+        make lint
+        make release
+        make real-clean hugo/build
+    - name: "Verify that Hugo files were built"
+      run: |
+        ls -lhatr
+        ls -lhatr ./public
+    - name: "Upload Hugo site artifact"
+      uses: actions/upload-artifact@v2
+      with:
+        name: hugo-site
+        path: ./public
+    #hugo-cache-pull:
+    - name: "Get SHAs"
+      id: get_shas
+      run: |
+        echo "::set-output name=current_sha_short::$(git rev-parse --short HEAD)"
```

Comment:
Why are we not using the github envs?

Reply:
https://stackoverflow.com/questions/59810838/how-to-get-the-short-sha-for-the-github-workflow

For computing `current_sha_short` it seems just as easy to stick with the way it is as it is to parse `github.sha`.

For computing `previous_sha_short`, though, adding the `fetch-depth: 2` parameter to `actions:checkout@v2` and then `git rev-parse`ing that is the only option.

If you think it'd make things simpler, both `current_sha_short` and `previous_sha_short` could be computed once and then passed around as artifacts, as needed.

Reply:
Well, I ended up doing exactly this.